backend/generalhw.pm

```diff
@@ -161,14 +156,13 @@ sub start_serial_grab {
     my ($self) = @_;
 
     $self->{serialpid} = fork();
-    if ($self->{serialpid} == 0) {
-        setpgrp 0, 0;
-        open(my $serial, '>',  $self->{serialfile});
-        open(STDOUT,     ">&", $serial);
-        open(STDERR,     ">&", $serial);
-        exec($self->get_cmd('GENERAL_HW_SOL_CMD'));
-        die "exec failed $!";
-    }
+    return unless $self->{serialpid} == 0;
```

Comment:
unless == looks so python :)

Reply:
uh, and what would look "more perl-ish"?